hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/CRC32ByteBufferImpl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.hadoop.ozone.common;
+
+import org.xerial.snappy.pool.BufferPool;
+import org.xerial.snappy.pool.CachingBufferPool;
+
+import java.nio.ByteBuffer;
+import java.util.function.Consumer;
+import java.util.zip.CRC32;
+
+public class CRC32ByteBufferImpl implements ChecksumByteBuffer {
+
+  public static final BufferPool POOL = CachingBufferPool.getInstance();
+  private final CRC32 checksum;
+
+  public CRC32ByteBufferImpl(CRC32 checksum) {
+    this.checksum = checksum;
+  }
+
+  @Override
+  public void update(ByteBuffer buffer) {
+    if (buffer.hasArray()) {
+      checksum.update(buffer.array(), buffer.position() + buffer.arrayOffset(),
+          buffer.remaining());
+    } else {
```

Comment:
As `checksum.update()` accepts a buffer, why do you need to copy the incoming buffer contents into a new (cached) buffer and then pass that new buffer to the `checksum.update()`. Could we not pass `buffer` directly to `checksum.update`?

I think it would be worth testing the performance difference between this approach, which is copying the incoming buffer into a new buf